File: calculator.py
#
def add(x, y):
    return x + y

def subtract(x, y):
    return x - y

def multiply(x, y):
    return x * y

def divide(x, y):

    if y != 0:
        return x / y
    else:
        return "Error: Cannot divide by zero"
        
def modulo(x , y):
    return x % y
    

print("Select an operation:")
print("\n1. Addition")
print("\n2. Subtraction")
print("\n3. Multiplication")
print("\n4. Division")
print("\n5. Modulo")

choice = input("\nEnter your choice (1/2/3/4/5): ")

num1 = float(input("Enter the first number: "))
num2 = float(input("Enter the second number: "))

if choice == '1':
    print(num1, "+", num2, "=", add(num1, num2))
elif choice == '2':
    print(num1, "-", num2, "=", subtract(num1, num2))
elif choice == '3':
    print(num1, "*", num2, "=", multiply(num1, num2))
elif choice == '4':
    print(num1, "/", num2, "=", divide(num1, num2))
elif choice == '5':
    print(num1, "%",num2, "=", modulo(num1, num2))
else:
    print("Invalid input")

File: Calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args){

        Scanner scan = new Scanner(System.in);
        String[] option = {"1", "2", "3", "4","5"};
        int x , y ;
        String choice;
        System.out.println("1. Addition");
        System.out.println("2. Subtraction");
        System.out.println("3. Multiplication ");
        System.out.println("4. Division");
        System.out.println("5. Modulo");  

        while(true){
            System.out.println("Enter the mathematical calculation you want to perform (1/2/3/4/5):");
            choice = scan.next();

            if(choice.equals("1") || choice.equals("2") || choice.equals("3") || choice.equals("4") || choice.equals("5")){
                System.out.println("Enter the first number:");
                x = scan.nextInt();
                System.out.println("Enter the Second number:");
                y = scan.nextInt();
                break;
            }
            System.out.println("invalid option,please try again.");

        }
        if (choice.equals("1")){
            int sum = x + y;
            System.out.println( x + " + " + y +" = "+ sum);

        }
        else if (choice.equals("2")){
            int subtraction  = x - y;
            System.out.println( x + " - " + y +" = "+ subtraction);
                
        }
        else if (choice.equals("3")){
            int multiplication = x * y;
            System.out.println( x + " * " + y +" = "+ multiplication);
        }
        else if (choice.equals("4")){
            if (y != 0){
            double division = x / y;
            System.out.println( x + " / " + y +" = "+ division);
            }else {
                System.out.println("Cannot divide by zero.");
            }
        }
        else if (choice.equals("5")){
            int modulo = x % y;
            System.out.println( x + " % " + y +" = "+ modulo);
        }
        System.out.println("Wanna calculate again");
        String calculateAgain = scan.nextLine();
        if(!calculateAgain.equals("Y")){
            //break;
        }
        scan.close();
    }
}
   